src/utils/script-setup.ts

```diff
@@ -171,7 +171,7 @@ function processDefineModel(node: Expression, context: Context): string | undefi
     return
   }
 
-  const model = context.utils.inferRuntimeType(context.ctx, modelTypeDecl)
+  const model = context.utils.inferRuntimeType(context.ctx, modelTypeDecl).filter(i => i !== 'Unknown')
   if (!model || model.length === 0 || model[0] === 'Unknown') {
     return
   }
```

Comment:
Vue handle like this: [vuejs/core/packages/compiler-sfc/src/script/defineModel.ts#L133-L140](https://github.com/vuejs/core/blob/4253a57f1703a7f1ac701d77e0a235689203461d/packages/compiler-sfc/src/script/defineModel.ts#L133-L140)

so the result should be `type: null`

[vue playground](https://play.vuejs.org/#eNp9kU9PwzAMxb9K5BNIYxWC09RNArQDSPwRcMyldF6XkSZR4oxKpd8dJ9XGDtNu8XvPzs9JD3fOTXcRYQZlqL1yJAJSdEJXpplLoCBhIU1tTSDRhkbMxQrXyuCzXaEuTWy/0ItfEc0orxYXl9KUxTiLO7kgbJ2uCLkSotxcL/o+jxqGsuAqq8q4SGJ31aaxfC/7EkTBZlkc9cOEiRhmrZrpNljD2H3ql1Db1imN/tWRYlgJM5Gd5FVa25+nrJGPONnr9Qbr7xP6NnRJk/DmMaDfoYSDR5VvkEZ7+fGCHZ8PJtNHzekz5jsGq2NiHGP3/HKMfZTLtI+ts56UaT7DsiM0Yb9UAk3JIecl8N89nFn9H/dmepv7pBlg+AP4YabG)

```suggestion

  let model = context.utils.inferRuntimeType(context.ctx, modelTypeDecl)
  const hasBoolean = model.includes('Boolean')
  const hasFunction = model.includes('Function')
  const hasUnknownType = model.includes('Unknown')
  if (hasUnknownType) {
    if (hasBoolean || hasFunction) {
      model = model.filter(t => t !== 'Unknown')
    }
    else {
      model = ['null']
    }
  }
  if (!model || model.length === 0) {
    return
  }
```